pkg/sypgp/sypgp.go

```diff
@@ -703,112 +702,108 @@ func getEncryptionAlgorithmName(n string) (string, error) {
 func date(s string) (string, error) {
 	if s == "" {
-		return "NULL", nil
+		return "[ultimate]", nil
 	}
 	if s == "none" {
-		return s + "\t\t\t", nil
+		return s, nil
 	}
 	c, _ := strconv.ParseInt(s, 10, 64)
-	ret := fmt.Sprintf("%v", time.Unix(c, 0))
+	ret := time.Unix(c, 0).String()
 
 	return ret, nil
 }
 
-// reformatMachineReadableOutput reformats the key search output that is in machine readable format
+// formatMROutputLongList reformats the key search output that is in machine readable format
 // see the output format in: https://tools.ietf.org/html/draft-shaw-openpgp-hkp-00#section-5.2
-func reformatMachineReadableOutput(keyText string) (string, error) {
-	var output = "\n\t\tFINGERPRINT\t\tALGORITHM  SIZE (BITS)\t\t  CREATION DATE\t\t\tEXPIRATION DATE\t\t\tSTATUS\t\t\tNAME/EMAIL" + "\n"
+func formatMROutputLongList(mrString string) (int, []byte, error) {
+	listLine := "%s\t%s\t%s\t%s\t%s\t%s\t%s\n"
 
-	rePubkey := regexp.MustCompile("pub:(.*)\n")
-	keys := rePubkey.FindAllString(keyText, -1)
-
-	var featuresKey []string
-
-	// for every key obtain the features: fingerprint, algorithm, size, creation date, expiration date, status and user(email)
-	for _, key := range keys {
-		var emailList = ""
-
-		detailsKey := strings.Split(key, ":")
-
-		for _, detail := range detailsKey {
-			d := strings.TrimSpace(detail)
-			if d != "" {
-				featuresKey = append(featuresKey, d)
-			}
-		}
-
-		fingerprint := featuresKey[1]
-
-		// regular expression to obtain the fingerprint of every key
-		reFingerprint := regexp.MustCompile("(" + fingerprint + ")[\\s+\\S]+?(::(\npub|\\s+$))")
-		emails := reFingerprint.FindAllString(keyText, -1)
-
-		// regular expression to obtain the email or emails from every key
-		reFormatEmail := regexp.MustCompile("uid:" + "\\w(.)+::")
-		userEmails := reFormatEmail.FindAllString(emails[0], -1)
-
-		for _, email := range userEmails {
-			detailsEmail := strings.Split(email, ":")
-
-			if emailList != "" {
-				emailList = emailList + "\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t" + detailsEmail[1] + "\n"
-			} else {
-				emailList = detailsEmail[1] + "\n"
-			}
-
-		}
-
-		algorithm, err := getEncryptionAlgorithmName(featuresKey[2])
-		if err != nil {
-			return "", err
-		}
-
-		size := featuresKey[3]
-
-		if len(size) < 4 {
-			size = size + " "
-		}
-		creationDate := featuresKey[4]
-		expiryDate := "none"
-
-		status := "enabled"
+	retList := bytes.NewBuffer(nil)
+	tw := tabwriter.NewWriter(retList, 0, 0, 2, ' ', 0)
+	fmt.Fprintf(tw, listLine, "FINGERPRINT", "ALGORITHM", "BITS", "CREATION DATE", "EXPIRATION DATE", "STATUS", "NAME/EMAIL")
 
-		if len(featuresKey) >= 6 {
-			expiryDate = featuresKey[5]
-		}
+	count := 0
+	keyNum := 0
 
-		creationTimestamp, err := date(creationDate)
-		if err != nil {
-			return "", err
-		}
+	key := strings.Split(mrString, "\n")
 
-		expirationTimestamp, err := date(expiryDate)
-		if err != nil {
-			return "", err
-		}
+	var (
```

Comment:
Why declare all of these variables here when they could be within the scope of the first loop? Then you would not need to "zero out" their values after iteration on the second loop completes.

Reply:
That wont work, the second loop can loop 1-3 times, and we dont know how much until the `uid` section.